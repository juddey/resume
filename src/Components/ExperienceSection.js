import React from 'react'
// import PropTypes from 'prop-types'
// import { FormattedMessage, injectIntl } from 'react-intl'
// import messages from './Messages/ExperienceSectionMessages'
import styles from './Styles/ExperienceSectionStyle'

export default class ExperienceSection extends React.Component {
  //  constructor (props) {
  //    super(props)
  //    this.state = {
  //      something: props.something
  //    }
  //  }

  // componentWillReceiveProps (newProps) {
  //   this.setState({ something: newProps.something })
  // }

  render () {
    return (
      <div style={styles.container}>
        <div>ExperienceSection Component</div>
      </div>
    )
  }
}

// react-intl Messages, outside of FormatMessage
// https://stackoverflow.com/questions/35186297/how-to-retrieve-a-string-in-reactintl-2-0-without-using-formattedmessage/39161208#39161208
// If you can't use <FormattedMessage /> for some reason, you can use this
// syntax: `this.context.intl.formatMessage({ ...messages.default })`
// by uncommenting the next block.
//  ExperienceSection.contextTypes = {
//   intl: PropTypes.object.isRequired
// }

// // Prop type warnings
// ExperienceSection.propTypes = {
//   someProperty: PropTypes.object,
//   someSetting: PropTypes.bool.isRequired
// }
//
// // Defaults for props
// ExperienceSection.defaultProps = {
//   someSetting: false
// }
