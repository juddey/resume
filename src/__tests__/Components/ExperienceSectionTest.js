import React from 'react'
import {
  createComponentWithIntl,
  shallowWithIntl
} from '../../../src/Support/Utils'
import ExperienceSection from '../../Components/ExperienceSection'

const div = document.createElement('div')
const wrapper = shallowWithIntl(<ExperienceSection />)

test('component exists', () => {
  expect(wrapper.length).toBe(1)
})

test('component structure', () => {
  expect(wrapper.name()).toEqual('div')
  expect(wrapper.children().length).toEqual(1)
})

test('renders to snapshot', () => {
  const tree = createComponentWithIntl(<ExperienceSection />, div).toJSON()
  expect(tree).toMatchSnapshot()
})
