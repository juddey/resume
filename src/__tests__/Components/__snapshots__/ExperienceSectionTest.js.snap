// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders to snapshot 1`] = `
<div
  style={
    Object {
      "flex": 1,
      "paddingTop": 15,
    }
  }
>
  <div>
    ExperienceSection Component
  </div>
</div>
`;
